### PR TITLE
[CAL-1222] Timezone dropdown shows london several times

### LINKED ISSUE #7660: [CAL-1222] Timezone dropdown shows london several times
### Issue Summary

When you search 'London' in the timezone dropdown it shows 3 times 'London GMT' but two of them are wrong. Two of them are timezone -5 and when clicking them it shows America/Detroit 

Screenshot 2023-03-10 at 10 16 13

<sub>[CAL-1222](https://linear.app/calcom/issue/CAL-1222/timezone-dropdown-shows-london-several-times)</sub>

### PR DESCRIPTION
**Pull request by gitstart-calcom**
## What does this PR do?

Fixes #7660 

https://www.loom.com/share/d8166e2baf1241f3aa6fb3a856e1d769

**Environment**: Staging(main branch) / Production

### CODE REVIEW

File: packages/trpc/server/routers/viewer.tsx [hunk 1]
```diff
@@ -158,7 +158,21 @@ const publicViewerRouter = router({
...
+    const maxPops: { [key: string]: { city: string; timezone: string; pop: number } } = {};
+    cityMapping.forEach((city) => {
+      const cityPop = city.pop;
+      if (maxPops[city.city]?.pop === undefined || cityPop > maxPops[city.city].pop) {
+        maxPops[city.city] = { city: city.city, timezone: city.timezone, pop: city.pop };
+      }
+    });
+    const uniqueCities = Object.values(maxPops);
+    uniqueCities.forEach((city) => {
+      if (city.city === "London") city.timezone = "Europe/London";
```
Comment (PeerRich) [inline packages/trpc/server/routers/viewer.tsx:171]: if we find more cities, can we add them here?

**Review by PeerRich — APPROVED** [review]
LGTM but waiting for @zomars

File: packages/trpc/server/routers/viewer.tsx (hunk 1 above)
Comment (gitstart-calcom) [reply packages/trpc/server/routers/viewer.tsx:172]: Yes

File: packages/trpc/server/routers/viewer.tsx [hunk 2]
```diff
@@ -159,7 +159,21 @@ const publicViewerRouter = router({
...
   // REVIEW: This router is part of both the public and private viewer router?
   slots: slotsRouter,
-  cityTimezones: publicProcedure.query(() => cityMapping),
+  cityTimezones: publicProcedure.query(() => {
+    const maxPops: { [key: string]: { city: string; timezone: string; pop: number } } = {};
+    cityMapping.forEach((city) => {
+      const cityPop = city.pop;
+      if (maxPops[city.city]?.pop === undefined || cityPop > maxPops[city.city].pop) {
+        maxPops[city.city] = { city: city.city, timezone: city.timezone, pop: city.pop };
+      }
```
Comment (zomars) [inline packages/trpc/server/routers/viewer.tsx:168]: Can we add some explainers on what's happening here? It is kinda difficult to tell what's going on in the code.

Comment (gitstart-calcom) [reply packages/trpc/server/routers/viewer.tsx:168]: We are filtering out all cities that have the same "city" key and only using the one with the highest population. This way we return a new array of cities without running the risk of having more than one city with the same name on the dropdown and prevent users from mistaking the time zone of the desired city

Comment (zomars) [reply packages/trpc/server/routers/viewer.tsx:168]: Can we put that in the code comments?

File: packages/trpc/server/routers/viewer.tsx [hunk 3]
```diff
@@ -159,7 +159,21 @@ const publicViewerRouter = router({
     }),
   // REVIEW: This router is part of both the public and private viewer router?
   slots: slotsRouter,
-  cityTimezones: publicProcedure.query(() => cityMapping),
+  cityTimezones: publicProcedure.query(() => {
+    const maxPops: { [key: string]: { city: string; timezone: string; pop: number } } = {};
+    cityMapping.forEach((city) => {
+      const cityPop = city.pop;
```
Comment (zomars) [inline packages/trpc/server/routers/viewer.tsx:165]: Let's try being more explicit here:

```suggestion
      const cityPopulationCount = city.pop;
```

**Review by zomars — APPROVED** [review]
Lazy loaded `city-timezones` (thanks @roae for the tip), added comments and improved legibility a bit.

Thank you for your contribution 🙏

**Pull request merged**